tests/surrogate_test.py

```diff
@@ -0,0 +1,206 @@
+    assert surrogate_object._coefficients is None
+    assert surrogate_object._grid is None
+    grid_object = surrogate_object.grid
+    assert isinstance(grid_object, IndexGrid)
+    assert surrogate_object.dimension == 1
+
+
+def test_initialization_2d():
+    """Test if class is properly intiallized."""
+    grid_generator = SmolyakGridGenerator(
+        ChebyshevFirstKind.make_nested_set(2))
+    domain = [(-5, 5), (-8, 0)]
+    grid_generator_ = 'not an IndexGridGenerator'
+    surrogate_object = Surrogate(domain, grid_generator)
+    assert numpy.allclose(surrogate_object.domain,
+                          numpy.array(domain, ndmin=2))
+    with pytest.raises(TypeError):
+        Surrogate(domain, grid_generator_)
+    assert surrogate_object._coefficients is None
+    grid_object = surrogate_object.grid
+    assert isinstance(grid_object, IndexGrid)
+    assert surrogate_object.dimension == 2
+
+
+def test_map_function_1d():
+    """Test if points are transfromed properly."""
+    grid = SmolyakGridGenerator(ChebyshevFirstKind.make_nested_set(1))
+    domain = (-8, 12)
+    new_domain = (0, 1)
+    f = Surrogate(domain, grid)
+    assert f._mapdomain(2, domain, new_domain) == pytest.approx(0.5)
+    assert numpy.allclose(f._mapdomain([-3, 7],
+                                       domain, new_domain), [0.25, 0.75])
+
+
+def test_map_function_2d():
+    """Test if points are transfromed properly."""
+    grid = SmolyakGridGenerator(ChebyshevFirstKind.make_nested_set(1))
+    domain = ((-10, 10), (0, 2))
+    new_domain = ((0, 1), (-1, 1))
+    f = Surrogate(domain, grid)
+    assert numpy.allclose(f._mapdomain((-10, 0), domain, new_domain), [0, -1])
+    assert numpy.allclose(f._mapdomain([(0, 1), (10, 2)],
+                                       domain, new_domain), [[0.5, 0], [1, 1]])
+
+
+def test_basis_matrix():
+    """Test if a correct basis matrix is generated."""
+    grid_generator = SmolyakGridGenerator(
+        ChebyshevFirstKind.make_nested_set(1))
+    surrogate_object = Surrogate([(-7, 3), (-18, 9)], grid_generator)
+    expected_basis_matrix = numpy.array(
+        [[1, 0, -1, 0, -1], [1, -1, 1, 0, -1], [1, 1, 1, 0, -1],
+         [1, 0, -1, -1, 1], [1, 0, -1, 1, 1]])
+    assert numpy.allclose(expected_basis_matrix,
+                          surrogate_object._make_basis_matrix())
+
+
+def test_surrogate_0():
+    """Test if surrogate generates exact results for test fucntion 0."""
+    surrogate_object = Surrogate([(-1, 1), (-1, 1), (-1, 1)],
+                                 SmolyakGridGenerator(
+                                     ChebyshevFirstKind.make_nested_set(2)))
+    surrogate_object.train(function_0, 'lu')
+    test_point = (0.649, 0, -0.9)
+    test_point_ = (-0.885, 1, 0.275)
+    assert numpy.allclose(surrogate_object(test_point),
+                          function_0([0.649, 0, -0.9]))
+    assert numpy.allclose(surrogate_object(test_point_),
+                          function_0([-0.885, 1, 0.275]))
```

Comment:
This would be much more readable, try applying this to your other tests:
```suggestion
    grid_gen = SmolyakGridGenerator(ChebyshevFirstKind.make_nested_set(2))
    surrogate = Surrogate([(-1, 1), (-1, 1), (-1, 1)], grid_gen)
    surrogate.train(function_0, 'lu')

    points = [(0.649, 0, -0.9), (-0.885, 1, 0.275)]
    surrogate_values = [surrogate(x) for x in points]
    exact_values = [function_0 for x in points]
    assert numpy.allclose(surrogate_values, exact_values)
```
Also, is there a reason that you chose these specific points? They seem weird, so a comment would be helpful.

Reply:
I just picked some random points from the domain.